src/core/Math.re

```diff
@@ -19,6 +19,23 @@ let lerp = (~acc, ~target, ~roundness) =>
   lerpf(~acc=Utils.foi(acc), ~target=Utils.foi(target), ~roundness)
   |> Utils.iof;
 
+// Map an input range to an output domain.
+let remapf = (~value, ~low1, ~high1, ~low2, ~high2) =>
+  low2 +. (high2 -. low2) *. ((value -. low1) /. (high1 -. low1));
+let remap = (~value, ~low1, ~high1, ~low2, ~high2) =>
+  remapf(
+    ~value=Utils.foi(value),
+    ~low1=Utils.foi(low1),
+    ~high1=Utils.foi(high1),
+    ~low2=Utils.foi(low2),
+    ~high2=Utils.foi(high2),
+  )
+  |> Utils.iof;
+
+// Normalization.
+let norm = (~value, ~low, ~high) =>
+  remapf(~value, ~low1=low, ~high1=high, ~low2=0., ~high2=1.);
```

Comment:
Beginning work for interpolation. `remapf` will allow us to do numeric interpolation for a specific range, and `norm` is just a shortcut to do `interpolation` to a 0-1 range.